### PR TITLE
Add a daily cloudwatch metrics report that's emailed to us. 

### PR DESCRIPTION
**Pull request by rahilb**
(at 9AM every morning). 

Example email: 

```
Identity CloudWatch Metrics for Sun Jul 15 2018 13:00:12 GMT+0000 (UTC) to Mon Jul 16 2018 13:00:12 GMT+0000 (UTC)
----------------------------------------------------------
SuccessfulRegistrations/SuccessfulRegistration: 534
SuccessfulSignIns/SuccessfulSignIn: 5432.000025921876
SuccessfulSignIns/SuccessfulSignInFirstStep: 2426
EmailsSent/identity-public-api/Guardian_account: 241
EmailsSent/identity-public-api/consent_mail: 368
EmailsSent/identity-public-api/password_reset: 90
EmailsSent/identity-public-api/password_set: 29
ExactTargetClient/identity-public-api/exactTarget.subscribeUserToList-success-count: 523
ExactTargetClient/identity-public-api/exactTarget.unsubscribeFromList-success-count: 74
```

Probably if we see any zero values we know something is really wrong 😄

### CODE REVIEW

File: node-lambdas/src/main/nodejs/identityDailyReport.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    const ctx = document.getElementById(metrics.Id).getContext('2d');
+    const chart = new Chart(ctx, {
+        type: 'bar',
+        data: {
+            labels: ["Red", "Blue", "Yellow", "Green", "Purple", "Orange"],
+            datasets: [{
+                label: '# of Votes',
+                data: [12, 19, 3, 5, 2, 3],
+                backgroundColor: [
+                    'rgba(255, 99, 132, 0.2)',
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/identityDailyReport.js]: does the email have colours? 👀

Comment (rahilb) [reply node-lambdas/src/main/nodejs/identityDailyReport.js]: not sure yet, just an experiment to make it pretty :)

Comment (walaura) [reply node-lambdas/src/main/nodejs/identityDailyReport.js]: ooh neat

File: node-lambdas/src/main/nodejs/identityDailyReport.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    });
+}
+
+function buildEmail(metricResults) {
+    const lines = [
+        `Identity CloudWatch Metrics for ${startTime} to ${endTime}`,
+        '----------------------------------------------------------'
+    ].concat(metricResults.map(d => {
+        return `${d.label}: ${d.value}`
+    }));
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/identityDailyReport.js]: we can turn this into a beautiful no-return arrow if you take out the braces and `return`.

```
concat(metricResults.map(d =>
    `${d.label}: ${d.value}`
));
```

**Comment by jfsoul** [comment]
Haven't had a chance to review this but how come this is in identity account deletion?

**Comment by rahilb** [comment]
>how come this is in identity account deletion?

This project already has all the riffraff boilerplate for auto deploying lambdas and updating cloudformation templates so it seemed like a logical place to go.

File: node-lambdas/package.json [hunk 3]
```diff
@@ -13,6 +13,7 @@
     "watch": "mocha -w \"src/test/nodejs/*.js\""
   },
   "dependencies": {
+    "asciichart": "^1.5.7",
```
Comment (QuarpT) [inline node-lambdas/package.json]: you can remove this since it's not used

**Comment by rahilb** [comment]
closing this, will send another PR in https://github.com/guardian/identity-daily-report

**Pull request closed without merging**